lightning/src/routing/router.rs

```diff
@@ -1380,6 +1435,27 @@ where L::Target: Logger {
 			return Err(LightningError{err: "Cannot route when there are no outbound routes away from us".to_owned(), action: ErrorAction::IgnoreError});
 		}
 	}
+	// Marshall blinded route hints
+	let mut blinded_route_hints = Vec::with_capacity(payment_params.payee.blinded_route_hints().len());
+	const DUMMY_BLINDED_SCID: u64 = 42;
+	for (idx, (blinded_payinfo, blinded_path)) in payment_params.payee.blinded_route_hints().iter().enumerate() {
+		if blinded_path.blinded_hops.len() == 1 || &blinded_path.introduction_node_id == our_node_pubkey {
```

Comment:
Note for reviewers: the 1-hop blinded path case is a bit weird since we'll totally ignore its `BlindedPayinfo` (and I don't think it's clear to what values 1-hop recipients should set payinfo).